Reformat code

rey opened the pull request:
This is a first pass of cleaning up the `scss` files in terms of formatting. I appreciate this is somewhat opinionated but has been done to standardise all the files in the repo and to improve readability :pray:
- Indents use (2) spaces only (previously there was a mixture of spaces and tabs)
- Strip trailing white space for sanity
- One line per declaration for readability
- Use `//` comment syntax instead of `/* */` for readability

sdougbrown commented:
I prefer tabs or 4 spaces, but this is all the cleanup that I meant to do a while ago - thanks.   I'll go through and double-check a bit later today. :smile:

rey commented:
@sdougbrown :thumbsup:

sdougbrown commented on _buttons.scss:
```diff
@@ -1,152 +1,183 @@
+}
+
+// if you click me, do I not bleed?
+.button:active, .button:focus {
```
gotcha - split to two lines :)

sdougbrown commented on _text.scss:
```diff
@@ -1,176 +1,370 @@
-/* careful with this... */
-.t-nat { font-size: 1rem; }
+// Element styling
+h1, h2, h3, h4, h5, h6, p, ul, ol, blockquote { padding-top: 0.5em; padding-bottom: 0.5em; }
```
this should probably be split into multiple lines as well

rey replied:
@sdougbrown d2edb746c215d03d2cc0256a7c30552174ac86b6 :+1:

sdougbrown commented on _text.scss:
```diff
@@ -1,176 +1,370 @@
+  .h-xxl--screen-m,
+  .t-xxxxl--screen-m,
+  .t-triple--screen-m {
+    font-size: 3em;
```
not your fault, but I just noticed in going through that we moved all of these sizes to vars, but not in the media queries.  oops.  we should probably update that!

sdougbrown commented:
Yeah I'm happy with the cleanup in general, but I don't know if I can live with the 2 spaces thing. :smile:   I personally prefer tabs in my CSS files, but there's some internal resistance to that and I feel like I'm losing the battle there.  Could we go to 4 spaces instead?

Just a heads-up, #2 is going to result in a few conflicts simply because some things are moving around - they should be easy to resolve.

rey commented:
> Yeah I'm happy with the cleanup in general, but I don't know if I can live with the 2 spaces thing.

I work with the `.sass` syntax (spits an error if you _don't_ use 2 spaces). That said, I'm quite flexible -- is there a common consensus amongst the @duckduckgo repos with regards to tabs/spaces/etc? It would be nice to be inline with other repos in the org.

> Just a heads-up, #2 is going to result in a few conflicts simply because some things are moving around - they should be easy to resolve.

I spotted that, shouldn't be too much of a head scratcher :smile:

sdougbrown commented:
>  is there a common consensus amongst the @duckduckgo repos with regards to tabs/spaces/etc?

Yeah I went looking and it turns out we have a code styleguide!  https://duck.co/duckduckhack/code_styleguide

It says that 4 spaces is the standard in _DuckDuckHack_ repos, and this is technically not a DuckDuckHack (i.e. Instant Answer) project so we're free to set our own standards but it'd probably be easier to go along with this. :smile:

sdougbrown commented:
I've created more conflicts for you in #5 based on the default vars conversation we had. :smile:

rey commented:
@sdougbrown
- Previously enforced a "one declaration per line" rule but since realised that this doesn't work on files like `_text.scss` and `_litestrap.scss` to the point where it harms readability. I believe this should be revisited at a later date -- for consistency right now, I've made all declarations inline across the repo.
- Change `_vars.scss` to `_variables.scss` for better readabilty.
- Change 2 spaces to 4 spaces (standard in DuckDuckHack repos).

sdougbrown commented:
It looks like you just added `_variables` instead of renaming the file.  I've done that before, just a git error.

Commit:
Further global refactoring

* Previously enforced a "one declaration per line" rule but since realised that
this doesn't work on files like _text.scss and _litestrap.scss to the point
where it harms readability. I believe this should be revisited at a later date
-- for consistency right now, I've made all declarations inline across the
repo.
* Change 2 spaces to 4 spaces (standard in DuckDuckHack repos).
* Add README.md

Commit:
Change _vars.scss to _variables.scss for better readabilty.

rey commented:
@sdougbrown :+1: https://github.com/rey/litestrap/commit/089ddc7d207ae647bea6c33ea103d8e87f3c2fd0

sdougbrown commented:
Thank you for the time you spent reformatting all of this.  I don't see any further issues, and I'm happy with where it's at.  :+1:
